src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteHandlers/AtMostOneFromBodyAttribute.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+using Microsoft.AspNetCore.App.Analyzers.Infrastructure;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteHandlers;
+
+using WellKnownType = WellKnownTypeData.WellKnownType;
+
+public partial class RouteHandlerAnalyzer : DiagnosticAnalyzer
+{
+    private static void AtMostOneFromBodyAttribute(
+        in OperationAnalysisContext context,
+        WellKnownTypes wellKnownTypes,
+        IMethodSymbol methodSymbol)
+    {
+        var fromBodyMetadataInterfaceType = wellKnownTypes.Get(WellKnownType.Microsoft_AspNetCore_Http_Metadata_IFromBodyMetadata);
+
+        var fromBodyMetadataInterfaceParameters = methodSymbol.Parameters.Where(p => p.HasAttributeImplementingInterface(fromBodyMetadataInterfaceType));
```

Comment:
Does this support the case where there's a single `[AsParameters]` parameter that contains two `[FromBody]`-decorated properties, or is that handled by a different analyser?

Reply:
This analyzer doesn't handle this case. Is that something you've come across in the wild?

Reply:
I haven't seen that in the wild no, but it seems like a hole in the analyser that it doesn't find incorrect usage of `[FromBody]` in all of the places its use is supported.

Reply:
OK, I've added support for the `[AsParameters]` attribute scenario. We may want to revisit the diagnostic message though.